org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/LocalGradleDistribution.java

```diff
@@ -32,11 +31,23 @@
     private final File location;
 
     LocalGradleDistribution(File location) {
-        this.location = Preconditions.checkNotNull(location);
-        Optional<String> error = GradleDistributionInfo.validateLocalInstallationLocation(location);
+        this.location = Preconditions.checkNotNull(location).getAbsoluteFile();
```

Comment:
Calling `getAbsoluteFile` does't really make sense this deep down in the code. If we can't deal with relative files, we should just assert that it's absolute. But actually I think we can deal with relative files just fine and shouldn't do anything surprising here.

Reply:
Changed to make it work with relative paths.